src/main/java/com/example/CinemaRoom/service/PurchaseService.java

```diff
@@ -1,42 +1,67 @@
 package com.example.CinemaRoom.service;
 
+import com.example.CinemaRoom.dto.SeatResponse;
+import com.example.CinemaRoom.dto.TicketResponse;
 import com.example.CinemaRoom.exception.PurchaseException;
 import com.example.CinemaRoom.model.Seat;
 import com.example.CinemaRoom.model.Ticket;
-import java.util.*;
+import com.example.CinemaRoom.repository.SeatRepository;
+import com.example.CinemaRoom.repository.TicketRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
 
+@Service
 public class PurchaseService {
 
-    private StatisticsService statisticsService;
-    private Map<String, Seat> allTicketPurchased = Collections.synchronizedMap(new HashMap<>());
+    private SeatRepository seatRepository;
+    private TicketRepository ticketRepository;
 
-    public PurchaseService(StatisticsService statisticsService) {
-        this.statisticsService =  statisticsService;
+    @Autowired
+    private SeatsService seatsService;
+
+    @Autowired
+    public void CinemaRoomApplication(SeatRepository seatRepository,TicketRepository ticketRepository) {
+        this.seatRepository = seatRepository;
+        this.ticketRepository = ticketRepository;
     }
 
-    public Seat returnTicket(String token) {
-        if(allTicketPurchased.containsKey(token)) {
-            Seat seat = allTicketPurchased.get(token);
-            allTicketPurchased.remove(token);
-            statisticsService.registerReturn(seat.price());
-            return seat;
+    private void saveTicket(Seat seat) {
+        if (!ticketRepository.findTicketBySeat(seat).isEmpty()) {
+            throw new PurchaseException("The ticket has been already purchased!");
         } else {
-            throw new PurchaseException("Wrong token!");
+            ticketRepository.save(new Ticket(UUID.randomUUID().toString(), seat));
         }
     }
 
-    public Ticket purchaseSeat(Seat seat) {
-        if(allTicketPurchased.containsValue(seat)) {
-            throw new PurchaseException("The ticket has been already purchased!");
-        } else {
-            String token = UUID.randomUUID().toString();
-            allTicketPurchased.put(token, seat);
-            statisticsService.registerPurchase(seat.price());
-            return new Ticket(token, seat);
+    private Ticket findSeatByToken(String token) {
+        List<Ticket> ticket = ticketRepository.findAllById(Collections.singleton(token));
+        if(ticket.isEmpty()) {
+            throw new PurchaseException("Wrong token!");
         }
+        else {
+          return ticket.get(0);
+        }
+    }
+
+    private Ticket purchaseSeat(int row, int column) {
+        Seat seat = seatsService.findSeatByRowAndColumn(row, column);
+        saveTicket(seat);
+        return ticketRepository.findTicketBySeat(seat).get(0);
```

Comment:
You can modify `saveTicket` function to `saveAndReturnTicket` and use it like this:

```suggestion
       return saveAndReturnTicket(seat);
```

Check the `save` function documentation for reference https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#save(S)